rust/processor/src/models/nft_marketplace_models/marketplace_utils.rs

```diff
@@ -0,0 +1,795 @@
+            TokenStandard::V2.to_string()
+        } else {
+            TokenStandard::V1.to_string()
+        }
+    }
+
+    pub fn get_collection_name_truncated(&self) -> String {
+        truncate_str(&self.collection_name, NAME_LENGTH)
+    }
+
+    pub fn from_event(event: &Event, transaction_version: i64) -> anyhow::Result<Option<Self>> {
+        serde_json::from_str(&event.data)
+            .map(|inner| Some(inner))
+            .context(format!(
+                "CollectionMetadata from event with version {} failed! failed to parse data {:?}",
+                transaction_version, event.data
+            ))
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum MarketplaceEvent {
+    ListingFilledEvent(ListingFilledEvent),
+    ListingCanceledEvent(ListingCanceledEvent),
+    ListingPlacedEvent(ListingPlacedEvent),
+    CollectionOfferPlacedEvent(CollectionOfferPlacedEvent),
+    CollectionOfferCanceledEvent(CollectionOfferCanceledEvent),
+    CollectionOfferFilledEvent(CollectionOfferFilledEvent),
+    TokenOfferPlacedEvent(TokenOfferPlacedEvent),
+    TokenOfferCanceledEvent(TokenOfferCanceledEvent),
+    TokenOfferFilledEvent(TokenOfferFilledEvent),
+    AuctionBidEvent(AuctionBidEvent),
+}
+
+impl MarketplaceEvent {
+    pub const LISTING_FILLED_EVENT: &'static str = "events::ListingFilledEvent";
+    pub const LISTING_CANCELED_EVENT: &'static str = "ListingCanceledEvent";
+    pub const LISTING_PLACED_EVENT: &'static str = "ListingPlacedEvent";
+    pub const COLLECTION_OFFER_PLACED_EVENT: &'static str = "CollectionOfferPlacedEvent";
+    pub const COLLECTION_OFFER_CANCELED_EVENT: &'static str = "CollectionOfferCanceledEvent";
+    pub const COLLECTION_OFFER_FILLED_EVENT: &'static str = "CollectionOfferFilledEvent";
+    pub const TOKEN_OFFER_PLACED_EVENT: &'static str = "TokenOfferPlacedEvent";
+    pub const TOKEN_OFFER_CANCELED_EVENT: &'static str = "TokenOfferCanceledEvent";
+    pub const TOKEN_OFFER_FILLED_EVENT: &'static str = "TokenOfferFilledEvent";
+    pub const AUCTION_BID_EVENT: &'static str = "AuctionBidEvent";
+
+    pub fn get_events_set() -> HashSet<&'static str> {
+        let mut events = HashSet::new();
+        events.insert(Self::LISTING_FILLED_EVENT);
+        events.insert(Self::LISTING_CANCELED_EVENT);
+        events.insert(Self::LISTING_PLACED_EVENT);
+        events.insert(Self::COLLECTION_OFFER_PLACED_EVENT);
+        events.insert(Self::COLLECTION_OFFER_CANCELED_EVENT);
+        events.insert(Self::COLLECTION_OFFER_FILLED_EVENT);
+        events.insert(Self::TOKEN_OFFER_PLACED_EVENT);
+        events.insert(Self::TOKEN_OFFER_CANCELED_EVENT);
+        events.insert(Self::TOKEN_OFFER_FILLED_EVENT);
+        events.insert(Self::AUCTION_BID_EVENT);
+        events
+    }
+
+    pub fn from_event(data_type: &str, data: &str, txn_version: i64) -> Result<Option<Self>> {
+        match data_type {
+            x if x.ends_with(Self::LISTING_FILLED_EVENT) => {
```

Comment:
you can't do this b/c what if it's a different address.

Reply:
Yeah, so the idea i have is to gate the whole marketplace processor at the beginning. Only the contract_address we have on file will pass the gating. Though this is still TODO part - I still have to all all other marketplaces.

Reply:
We can early return and gate it here too